core/src/main/java/io/questdb/std/Chars.java

```diff
@@ -174,6 +174,28 @@ public static boolean containsLowerCase(@NotNull CharSequence sequence, @NotNull
         return indexOfLowerCase(sequence, 0, sequence.length(), termLC) != -1;
     }
 
+    public static boolean containsWord(CharSequence seq, CharSequence term, char separator) {
+        if (Chars.isBlank(seq)) {
+            return false;
+        }
+        if (Chars.isBlank(term)) {
+            return false;
+        }
+        int i = Chars.indexOf(seq, 0, seq.length(), term);
```

Comment:
nit: both sequences are immutable (for the duration of this method), length() doesn't change